app/code/Magento/Catalog/Model/Indexer/Category/Flat/AbstractAction.php

```diff
@@ -511,32 +521,15 @@ protected function getTableName($name)
         return $this->resource->getTableName($name);
     }
 
-    /**
-     * Get category metadata instance.
-     *
-     * @return EntityMetadata
-     */
-    private function getCategoryMetadata()
-    {
-        if (null === $this->categoryMetadata) {
-            $metadataPool = ObjectManager::getInstance()
-                ->get(\Magento\Framework\EntityManager\MetadataPool::class);
-            $this->categoryMetadata = $metadataPool->getMetadata(CategoryInterface::class);
-        }
-        return $this->categoryMetadata;
-    }
-
     /**
      * Get skip static columns instance.
      *
      * @return array
      */
     private function getSkipStaticColumns()
     {
-        if (null === $this->skipStaticColumns) {
-            $provider = ObjectManager::getInstance()
-                ->get(SkipStaticColumnsProvider::class);
-            $this->skipStaticColumns = $provider->get();
+        if ([] === $this->skipStaticColumns) {
+            $this->skipStaticColumns = $this->staticColumnsProvider->get();
```

Comment:
Please look comments above

```suggestion
        if ($this->skipStaticColumns === []) {
            $this->skipStaticColumns = $this->skipStaticColumnsProvider->get();
```